### ISSUE TITLE
Question: MeshAgent Extended Device/Group information

### ISSUE THREAD

**Issue by openmoto** [Ylianst/MeshCentral#950]
Hi @Ylianst ,

I'm not sure if this could be a security concern or even if it's possible as of now. 
Would it be possible to be able to query meshagent for device and device group information?

For example, if I have a HTML report I generate per computer with powershell.
Instead of creating a separate script per device group, I would then be able to query mesh central for the group name and possibly the email addresses of the users that have admin or management rights?

I'm not sure how the feature would be implemented, but one thought is:
We have General and Summary tab under My Account > Device Groups > Group
If a third tab "Customer Info" or something similar can be added where one can create information that can be requested through meshcentral.. encrypted credentials and other information could be stored there and retrieved through meshagent when running scripts.

**Comment by ryanblenis** [comment]
Hi @openmoto,

As a partial answer / suggestion for getting the group name for use in a script; You could parse the MeshAgent.msh file for the MeshName or MeshID and use a switch in PowerShell to set your variables.

**Comment by Ylianst** [comment]
On Windows, the MeshAgent will store a bunch of information under the "LOCAL_MACHINE\SOFTWARE\Open Source\MeshAgent2" key. Note that this registry path is a bit different if running the 32bit agent on a 64bit Windows machine. Information includes manu of the same as the .msh, but also the NodeID of the device. Let me know if this helps.

**Comment by openmoto** [comment]
Thanks Guys,
I looked at both options and it looks like the registry path stores the information encrypted.
As per @ryanblenis' suggestion, I've been able to import the information from MeshAgent.msh using powershell:
`$system = Get-Content -raw 'C:\Program Files\Mesh Agent\MeshAgent.msh' | ConvertFrom-StringData`

Then using `$system.MeshName` to return the Mesh Name.

With this I should be able to run the same script across windows, mac and linux and get similar results based on which group the computer belongs to.

Thanks
